queries/places.py

```diff
@@ -0,0 +1,352 @@
+    | "hvaða" "opnunartímar" "eru" QPlacesPrepAndSubject
+    | "hversu" "lengi" "er" "opið" QPlacesPrepAndSubject
+    | "hve" "lengi" "er" "opið" QPlacesPrepAndSubject
+    | "klukkan" "hvað" "opnar" QPlacesPrepAndSubject
+    | "klukkan" "hvað" "opnar" QPlacesSubjectNf
+    | "hvenær" "lokar" QPlacesPrepAndSubject
+    | "hvenær" "lokar" QPlacesSubjectNf
+    | "hvenær" "opnar" QPlacesPrepAndSubject
+    | "hvenær" "opnar" QPlacesSubjectNf
+    | "klukkan" "hvað" "lokar" QPlacesPrepAndSubject
+    | "klukkan" "hvað" "lokar" QPlacesSubjectNf
+    | "hversu" "langt" "er" "í" "lokun" QPlacesPrepAndSubject
+    | "hversu" "langt" "er" "í" "lokun" QPlacesSubjectEf
+    | "hvað" "er" "langt" "í" "lokun" QPlacesPrepAndSubject
+    | "hvað" "er" "langt" "í" "lokun" QPlacesSubjectEf
+    | "hvenær" "er" "opið" QPlacesPrepAndSubject
+    | "hvað" "er" QPlacesSubjectNf QPlOpen "lengi"
+    | "hve" "lengi" "er" QPlacesSubjectNf QPlOpen
+    | "hversu" "lengi" "er" QPlacesSubjectNf QPlOpen
+
+QPlacesIsOpen →
+    "er" "opið" QPlacesPrepAndSubject QPlNow?
+    | "er" QPlacesSubjectNf QPlOpen QPlNow?
+
+QPlacesIsClosed →
+    "er" "lokað" QPlacesPrepAndSubject QPlNow?
+    | "er" QPlacesSubjectNf QPlClosed QPlNow?
+
+QPlacesAddress →
+    "hvert" "er" "heimilisfangið" QPlacesPrepAndSubject
+    | "hvað" "er" "heimilisfangið" QPlacesPrepAndSubject
+    | "hvert" "er" "heimilisfang" QPlacesSubjectEf
+    | "hvað" "er" "heimilisfang" QPlacesSubjectEf
+    | "hvar" "er" QPlacesSubjectNf "til" "húsa"
+    | "hvar" "er" QPlacesSubjectNf "staðsett" 
+    | "hvar" "er" QPlacesSubjectNf "staðsettur"
+    | QPlacesPreposition "hvaða" "götu" "er" QPlacesSubjectNf
+    | QPlacesPreposition "hvaða" "stræti" "er" QPlacesSubjectNf
+
+QPlacesPrepAndSubject →
+    QPlacesPrepWithÞgf QPlacesSubjectÞgf
+    | QPlacesPrepWithÞf QPlacesSubjectÞf
+
+QPlacesSubjectNf →
+    Nl_nf
+
+QPlacesSubjectÞf →
+    Nl_þf
+
+QPlacesSubjectÞgf →
+    Nl_þgf
+
+QPlacesSubjectEf →
+    Nl_ef
+
+QPlacesPreposition →
+    QPlacesPrepWithÞf | QPlacesPrepWithÞgf
+
+QPlacesPrepWithÞgf →
+    "á" | "í" | "hjá"
+
+QPlacesPrepWithÞf →
+    "við" | "fyrir"
+
+QPlOpen →
+    "opið" | "opin" | "opinn"
+
+QPlClosed →
+    "lokað" | "lokuð" | "lokaður"
+
+QPlNow →
+    "núna" | "í" "augnablikinu" | "eins" "og" "stendur" | "nú"
+
+QPlToday →
+    "núna"? "í" "dag" | "núna"? "í" "kvöld"
+
+$score(+35) QPlacesQuery
+
+"""
+
+
+_PLACENAME_MAP = {}
+
+
+def _fix_placename(pn):
+    p = pn.capitalize()
+    return _PLACENAME_MAP.get(p, p)
+
+
+def QPlacesQuery(node, params, result):
+    result["qtype"] = _PLACES_QTYPE
+
+
+def QPlacesOpeningHours(node, params, result):
+    result["qkey"] = "OpeningHours"
+
+
+def QPlacesIsOpen(node, params, result):
+    result["qkey"] = "IsOpen"
+
+
+def QPlacesIsClosed(node, params, result):
+    result["qkey"] = "IsClosed"
+
+
+def QPlacesAddress(node, params, result):
+    result["qkey"] = "PlaceAddress"
+
+
+def QPlacesSubjectNf(node, params, result):
```

Comment:
This becomes just `QPlacesSubject(...)` if the nonterminal is changed to `QPlacesSubject/fall`